src/runs.c

```diff
@@ -90,6 +90,41 @@ r_obj* vec_identify_runs(r_obj* x) {
 
 // -----------------------------------------------------------------------------
 
+r_obj* ffi_vec_run_sizes(r_obj* x) {
+  return vec_run_sizes(x);
+}
+
+r_obj* vec_run_sizes(r_obj* x) {
```

Comment:
https://github.com/r-lib/vctrs/pull/1771/commits/c5f054c2a468508e69df43a1ce1a5b897eb55f4f adds a `vec_check_vector()` check in `vec_detect_run_bounds0()`, which is used by all run-related functions.

I've also now passed the `error_call` down through all of the run functions. This generates better error calls and will make future addition of an R level `error_call` arg easier if we decide we need that.

I think together that is "good enough" because if I call a `vec_*()` function that has an `error_call` argument (here, `vec_detect_run_bounds0()`) then I expect that it will eventually throw a good error on non-vectors and will use the `error_call` to show the right call location